### PR TITLE
Reduce number peaks

### PR DESCRIPTION
**Pull request by jspaaks**
Originally here https://github.com/matchms/matchms-backup/pull/209

### CODE REVIEW

File: matchms/filtering/reduce_to_number_of_peaks.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    n_required: int
+        Number of minimum required peaks. Spectra with fewer peaks will be set
+        to 'None'. Default is 1.
+    n_max: int
+        Maximum number of peaks. Remove peaks if more peaks are found.
+    ratio_desired: float, optional
+        Set desired ratio between maximum number of peaks and parent mass.
+        Default is None.
+    """
+    def _set_maximum_number_of_peaks_to_keep(spectrum, ratio_desired):
```
Comment (jspaaks) [inline matchms/filtering/reduce_to_number_of_peaks.py]: the pythonic way is to not have these arguments, just have 
```python
def _set_maximum_number_of_peaks_to_keep():
```

File: matchms/filtering/reduce_to_number_of_peaks.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        parent_mass = spectrum.get("parent_mass", None)
+        if parent_mass and ratio_desired:
+            n_desired_by_mass = int(ceil(ratio_desired * parent_mass))
+            threshold = max(n_required, n_desired_by_mass)
```
Comment (jspaaks) [inline matchms/filtering/reduce_to_number_of_peaks.py]: ``threshold`` shadows a name from outer scope, can be avoided by just ``return``ing without assignment first

File: matchms/filtering/reduce_to_number_of_peaks.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        else:
+            threshold = n_max
+        return threshold
+
+    def _remove_lowest_intensity_peaks(spectrum):
```
Comment (jspaaks) [inline matchms/filtering/reduce_to_number_of_peaks.py]: the pythonic way is to not have this argument, just have
```python
def _remove_lowest_intensity_peaks():
```

File: matchms/filtering/reduce_to_number_of_peaks.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        mz, intensities = spectrum.peaks
+        idx = intensities.argsort()[-threshold:]
+        idx_sort_by_mz = mz[idx].argsort()
+        spectrum.peaks = Spikes(mz=mz[idx][idx_sort_by_mz],
```
Comment (jspaaks) [inline matchms/filtering/reduce_to_number_of_peaks.py:34]: I think we should look into adding ``getitem`` and ``setitem`` to ``Spikes``, so we can avoid these ugly indices (but let's do that later).

File: matchms/filtering/reduce_to_number_of_peaks.py (hunk 3 above)
Comment (florian-huber) [reply matchms/filtering/reduce_to_number_of_peaks.py]: Done.

File: matchms/filtering/reduce_to_number_of_peaks.py (hunk 1 above)
Comment (florian-huber) [reply matchms/filtering/reduce_to_number_of_peaks.py]: Done.

File: matchms/filtering/reduce_to_number_of_peaks.py (hunk 2 above)
Comment (florian-huber) [reply matchms/filtering/reduce_to_number_of_peaks.py]: Agreed, that looks better.

File: matchms/filtering/reduce_to_number_of_peaks.py (hunk 4 above)
Comment (florian-huber) [reply matchms/filtering/reduce_to_number_of_peaks.py:34]: Yes, a setter for Spikes would be nice!
(to be done later)

**Comment by florian-huber** [comment]
Thanks for the comments @jspaaks .  I changed the code accordingly. Let me know if that looks OK for you then I can merge it.

File: matchms/filtering/reduce_to_number_of_peaks.py (hunk 4 above)
Comment (jspaaks) [reply matchms/filtering/reduce_to_number_of_peaks.py:34]: https://github.com/matchms/matchms/issues/43

**Comment by jspaaks** [comment]
@florian-huber, this is good to merge. GitHub won't let me approve at the moment because I transferred this issue from matchms-backup repo, which made me the author.

**Comment by florian-huber** [comment]
Thanks Juriaan. I'll merge it.

**Pull request merged**